inventory/kits/__tests__/api/test_api_retire_kit.py

```diff
@@ -0,0 +1,137 @@
+            name="Test Blueprint",
+            complete_content=self.compressed_blueprint_content,
+        )
+
+    def create_items(self):
+        self.item = Item.objects.create(
+            name="Another Item",
+            type="General",
+            unit="units",
+            total_quantity=100,
+            min_quantity=10,
+            is_opened=False,
+        )
+        self.itemExpiry1 = self.item.expiry_dates.create(
+            expiry_date="2023-12-31", quantity=50, archived=False
+        )
+        self.itemExpiry2 = self.item.expiry_dates.create(
+            expiry_date="2024-12-31", quantity=50, archived=False
+        )
+
+        self.item_no_expiry = Item.objects.create(
+            name="No Expiry Item",
+            type="General",
+            unit="units",
+            total_quantity=100,
+            min_quantity=10,
+            is_opened=False,
+        )
+        self.itemExpiry_no_expiry = self.item_no_expiry.expiry_dates.create(
+            quantity=50, archived=False
+        )
+
+    def clear_relevant_models(self):
+        History.objects.all().delete()
+        Kit.objects.all().delete()
+        Blueprint.objects.all().delete()
+        Item.objects.all().delete()
+
+    def test_retire_kit(self):
+        order_count = Order.objects.count()
+        response = self.client.get(
+            reverse("retire_kit", args=[self.kit_id]), None, format="json"
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.data["message"], "Kit retired and contents deposited successfully!"
+        )
+
+        # check that kit is retired
+        kit = Kit.objects.get(id=self.kit_id)
+        self.assertEqual(kit.status, "RETIRED")
+
+        # check that history is created
+        history = History.objects.get(kit=kit)
+        self.assertEqual(history.type, "RETIREMENT")
+
+        # check that items are deposited
+        item_expiry = ItemExpiry.objects.get(id=self.item_expiry1_id)
+        self.assertEqual(item_expiry.quantity, 55)
+        item_expiry = ItemExpiry.objects.get(id=self.item_expiry2_id)
+        self.assertEqual(item_expiry.quantity, 55)
+        item_expiry = ItemExpiry.objects.get(id=self.item_no_expiry_id)
+        self.assertEqual(item_expiry.quantity, 55)
+
+        # Check that an order is created
+        self.assertEqual(Order.objects.count(), order_count + 1)
+        order = Order.objects.get(id=response.data["order_id"])
+        self.assertEqual(order.reason, "kit_retire")
+        order_items = OrderItem.objects.filter(order=order)
+        self.assertEqual(order_items.count(), 3)
+        items_ordered = [
+            self.item_expiry1_id,
+            self.item_expiry2_id,
+            self.item_no_expiry_id,
+        ]
+        for order_item in order_items:
+            self.assertEqual(order_item.item_expiry.id, items_ordered.pop(0))
+            self.assertEqual(order_item.ordered_quantity, 5)
```

Comment:
The field asserted here should be order_item.returned_quantity? It returns None in this test case but the correct value is reflected in order_item.ordered_quantity instead. Can look into this issue and update if it is of concern? @jonasgwt

Reply:
should be correct `returned_quantity` is used for loans only